packages/design-system/src/components/tooltip.tsx

```diff
@@ -85,6 +99,102 @@ export const Tooltip = React.forwardRef(function TooltipWrapper(
   );
 });
 
+/**
+ * EnhancedTooltip has the following differences from the radix-ui
+ * 1. Don't show the tooltip if any click or key-down was made inside Tooltip.Trigger
+ * 2. Show Tooltip on focus after the delay
+ **/
+export const EnhancedTooltip = React.forwardRef(
+  function TextControlTooltipWrapper(
+    props: Omit<TooltipProps, "open" | "onOpenChange">,
+    ref: Ref<HTMLDivElement>
+  ) {
+    const [open, setOpen] = useState(props.defaultOpen ?? false);
+
+    /**
+     * Used to skip multiple focus/blur calls
+     */
+    const isFocusedRef = useRef(false);
+
+    /**
+     * Disable opening if any interaction with control occurred
+     */
+    const allowOpenRef = useRef(true);
+
+    const handleDisableOpen = () => {
+      allowOpenRef.current = false;
+      setOpen(false);
+      showTooltipDelayed.cancel();
+    };
+
+    const showTooltipDelayed = useDebouncedCallback(
+      () => {
+        if (allowOpenRef.current) {
+          setOpen(true);
+        }
+      },
+      props.delayDuration ?? 700,
+      { leading: false }
+    );
+
+    // We debounce blur/focus events in a single function, to skip fast `blur/focus` which can occur during select menu hovers
+    const handleFocusEvents = useDebouncedCallback(
+      (eventType: "focus" | "blur") => {
+        if (eventType === "blur") {
+          isFocusedRef.current = false;
+          allowOpenRef.current = true;
+          setOpen(false);
+          showTooltipDelayed.cancel();
+          return;
+        }
+
+        if (isFocusedRef.current === false) {
+          showTooltipDelayed();
+        }
+
+        isFocusedRef.current = true;
+      },
+      0,
+      { leading: false }
+    );
+
+    const triggerProps = {
+      onFocus: (event: FocusEvent<HTMLElement>) => {
+        handleFocusEvents("focus");
+        event.preventDefault();
+      },
+      onBlur: (event: FocusEvent<HTMLElement>) => {
+        handleFocusEvents("blur");
+        event.preventDefault();
+      },
+      onPointerDown: () => {
+        handleDisableOpen();
+      },
+      onKeyDown: () => {
+        handleDisableOpen();
+      },
+    };
+
+    const onOpenChange = (isOpen: boolean) => {
+      if (allowOpenRef.current) {
+        setOpen(isOpen);
+      }
+    };
+
+    // Wrap in provider so each control will use own hover delay
+    return (
+      <TooltipPrimitive.Provider>
```

Comment:
can't we have a provider used separately wrapped around entire style panel? so we can set the settings for timeout on it in a single place?